src/Neo.CLI/CLI/MainService.Tools.cs

```diff
@@ -445,6 +571,214 @@ private string ClearHexString(string hexString)
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Converts a Private key to neo3 address
+        /// </summary>
+        /// <param name="wif">Private key in WIF format</param>
+        /// <returns>Returns null when the string does not represent a private key; otherwise
+        /// returns the string that represents the converted neo3 address.
+        /// </returns>
+        public static string? PrivateKeyToAddress(string wif)
+        {
+            try
+            {
+                var pubKey = PrivateKeyToPublicKey(wif);
+                return Contract.CreateSignatureContract(ECPoint.Parse(pubKey, ECCurve.Secp256r1)).ScriptHash.ToAddress(0x35);
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Converts a Scripts to OpCode
+        /// </summary>
+        /// <param name="base64">Scripts in Base64 string</param>
+        /// <returns>Returns null when the string does not represent a scripts; otherwise
+        /// returns the string that represents the converted OpCode.
+        /// </returns>
+        public string? ScriptsToOpCode(string base64)
+        {
+            List<byte> scripts;
+            try
+            {
+                scripts = Convert.FromBase64String(base64).ToList();
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+            try
+            {
+                _ = new Script(scripts.ToArray(), true);
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+            return ScriptsToOpCode(scripts);
+        }
+
+        /// <summary>
+        /// Converts a Hex Scripts to OpCode
+        /// </summary>
+        /// <param name="hex">Scripts in Hex string</param>
+        /// <returns>Returns null when the string does not represent a scripts; otherwise
+        /// returns the string that represents the converted OpCode.
+        /// </returns>
+        public string? HexScriptsToOpCode(string hex)
+        {
+            List<byte> scripts;
+            try
+            {
+                scripts = hex.HexToBytes().ToList();
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+            try
+            {
+                _ = new Script(scripts.ToArray(), true);
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+            return ScriptsToOpCode(scripts);
+        }
+
+        private string ScriptsToOpCode(List<byte> scripts)
+        {
+            //Initialize all OpCodes
+            var OperandSizePrefixTable = new int[256];
+            var OperandSizeTable = new int[256];
+            foreach (FieldInfo field in typeof(OpCode).GetFields(BindingFlags.Public | BindingFlags.Static))
```

Comment:
Why not use `Script` and GetInstruction?

Reply:
Yes you `Script` class or something like this
```
byte[] data = [];

foreach (var d in data)
{
    Console.WriteLine(Enum.GetName((OpCode)d));
}
```